docs/Contributing/pull-request.rst

```diff
@@ -0,0 +1,182 @@
+
+* File updates from `Upstream` only come "down" to the local `Sandbox` unidirectionally.
+* File updates from `Origin` to `Upstream` only happen from `Origin` to `Upstream`, unidirectionally.
+* File updates from the local `Sandbox` to `Origin` can occur bi-directionally. However, the most common way is
+  "up" from the `Sandbox` to `Origin`.
+
+What typically happens in a development cycle looks something like this: Upstream -> Sandbox -> Origin ->
+Upstream. With this in mind, submitting a Pull Request generally entails these steps:
+
+#. Update the Sandbox with the lastest updates/changes from Upstream.
+#. Ensure Origin is also up to date with Upstream.
+#. Create a local branch from Upstream.
+#. Edit files as needed in the Sandbox and local branch.
+#. Push the changes in a commit to Origin.
+#. Requesting a Pull Request from Origin to Upstream.
+
+
+1. Update The Sandbox With The Lastest Updates/Changes From Upstream
+====================================================================
+
+Assuming you are in the local Sandbox, issue these commands to get the latest code from Upstream:
+
+.. code-block:: rst
+
+  ~/GitHub/freeciv21$ git checkout master
+  ~/GitHub/freeciv21$ git fetch upstream master
+  ~/GitHub/freeciv21$ git pull upstream master --ff-only
+
+
+.. note::
+  If you want :code:`git pull` to always do a "fast-forward" (which is preferred), you can set a local
+  repository configuration option with this command: :code:`git config --local pull.ff only`. You only need to
+  do this once. After the config is set, you can drop the :code:`--ff-only` option from the last command above.
+
+
+2. Ensure Origin Is Also Up To Date With Upstream
+=================================================
+
+Assuming you are in the local Sandbox and have recently completed the 3 commands above, issue this command to
+get the latest code from Upstream pushed to your personal fork (Origin):
+
+.. code-block:: rst
+
+  ~/GitHub/freeciv21$ git push origin
+
+
+You will be prompted to enter your SSH passkey at this time.
+
+
+3. Create A Local Branch From Upstream
+======================================
+
+Now that things are all up to date with the lastest code, let's create a branch to do your work in.
+
+.. code-block:: rst
+
+  ~/GitHub/freeciv21$ git checkout -b [some_feature] upstream/master
+
+
+The tag :code:`[some_feature]` should be replaced by something that matches what you are planning to work on.
+For example, it could be as easy as :code:`issue_123`, meansing that this branch is changes to resolve GitHub
+Issue 123 in the Freeciv21 Issues list.
+
+.. attention::
+  Never make changes in the master branch, always make changes in a different branch. This makes it much
+  easier to evaluate changes as part of the Pull Request evaluation process and keeps things much cleaner
+  from a repository management perspective. This follows best practices and keeps the master branch from
+  being accidentially messed up.
```

Comment:
Don't make them afraid that they mess up *our* master branch by doing something wrong :D Actually we could say somewhere that they can't break anything on our side unless we approve it − in which case we well be guilty.

Reply:
Indeed. I feel the entire discussion here is a bit unnecessary (sorry jwrober). There are a bunch of manuals on the internet which repeat the same thing. 

And, not making changes in the master branch is something they will figure out themselves. We dont need to teach best git practices to people. We only care about the code. If there are pulls with a lot of bad commits.. We can handle in the pull request.

Reply:
Maybe instead of all this general git stuff we talk more about what we particularly do ? Like the way we do merges without keeping the merge commit etc and also what all github actions are available and what do we see when merging a commit.

Reply:
language made less "intense", emphasize best practice